lib/flight_job/models/script/metadata.rb

```diff
@@ -0,0 +1,117 @@
+# PARTICULAR PURPOSE. See the Eclipse Public License 2.0 for more
+# details.
+#
+# You should have received a copy of the Eclipse Public License 2.0
+# along with Flight Job. If not, see:
+#
+#  https://opensource.org/licenses/EPL-2.0
+#
+# For more information on Flight Job, please visit:
+# https://github.com/openflighthpc/flight-job
+#==============================================================================
+require 'json'
+require 'json_schemer'
+require_relative "../metadata/base_metadata"
+
+module FlightJob
+  class Script < ApplicationModel
+    # Encapsulates a script's `metadata.yaml` file.
+    #
+    # * Loads and saves the file.
+    # * Validates against a schema.
+    class Metadata < Metadata::BaseMetadata
+
+      SCHEMA = JSONSchemer.schema({
+        "$comment" => "strip-schema",
+        "type" => "object",
+        "additionalProperties" => false,
+        "required" => ['created_at', 'script_name'],
+        "properties" => {
+          # ----------------------------------------------------------------------------
+          # Required
+          # ----------------------------------------------------------------------------
+          'created_at' => { 'type' => 'string', 'format' => 'date-time' },
+          'script_name' => { 'type' => 'string' },
+          # ----------------------------------------------------------------------------
+          # Psuedo - Required
+          #
+          # These should *probably* become required on the next major release of the metadata
+          # ----------------------------------------------------------------------------
+          'answers' => { 'type' => 'object' },
+          'tags' => { 'type' => 'array', 'items' => { 'type' => 'string' }},
+          'template_id' => { 'type' => 'string' },
+          'version' => { 'const' => 0 },
+        }
+      })
+
+      validate do
+        schema_errors = SCHEMA.validate(self.to_hash).to_a
+        unless schema_errors.empty?
+          path_tag = File.exist?(@path) ? @path : @parent.id
+          FlightJob.logger.info("Invalid metadata: #{path_tag}\n")
+          JSONSchemaErrorLogger.new(schema_errors, :info).log
+          errors.add(:metadata, 'is not valid')
+        end
+      end
+
+      attributes \
+        :created_at,
+        :tags,
+        :template_id,
+        :script_name
+
+      attribute :answers, default: {}
+
+      def self.from_template(template, answers, script)
+        initial_metadata = {
+          "version" => 0,
+          "created_at" => Time.now.rfc3339,
+          "template_id" => template.id,
+          "script_name" => template.script_template_name,
+          "answers" => answers,
+          "tags" => template.tags,
+        }
+        new(initial_metadata, script.metadata_path, script)
+      end
+
+      def self.create_blank(path,parent)
+        initial_metadata = { "created_at" => Time.now.rfc3339 }
+        new(initial_metadata, path, parent)
+      end
```

Comment:
Discussed with @benarmston on slack - changing the name of `create_blank` but leaving the contents the same. `super` would call a class method so it would be awkward to edit the `@hash` instance variable it would generate.